code/core/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/HistoricalEventsQuerying.kt

```diff
@@ -48,3 +59,43 @@ internal fun historicalEventsQuerying(
         0
     }
 }
+
+/**
+ * Converts the result of an event history lookup query to an integer.
+ * For an "any" search (event order does not matter), the value returned will be the count of all matching events in EventHistory
+ * For an `.ordered` search (events must occur in the provided order), the value returned will be 1 if the events were found in the provided order, or 0 otherwise.
+ * If a database error occurred, this method will always return -1
+ *
+ * @param enforceOrder `boolean` if true, the events must have been recorded in the same order as the request
+ * @param eventHistoryResult `Array<EventHistoryResult>` the result of the event history lookup query
+ */
+@JvmSynthetic
+internal fun convertEventHistoryResultToInt(
+    enforceOrder: Boolean,
+    eventHistoryResult: Array<EventHistoryResult>
+): Int {
+    if (enforceOrder) {
+        for (result in eventHistoryResult) {
+            // Early exit on database error
+            if (result.count == -1) {
```

Comment:
0's and 1's here are a bit too involved for code readability. Can we add designated constants for database error, all events found , no records found etc and use them across the calling components?